Use most popular songs

Noutbuk opened the pull request.

Noutbuk committed:
Add new room

Perform cleanup

ansgarprause reviewed (approved):
<img src="https://media1.giphy.com/media/3o7TKDGVU5nYvrBiDu/giphy.gif"/>

Noutbuk commented:
Will rebase my code and reopen the PR from a branch.
Sadly, too many artists cause a timeout in `load_sample_tracks.js`.
Maybe wen can batch the requests or change the implementation s.t. you can directly add individual songs?

Noutbuk commented:
My basic idea was to a apple music playlist and extract artist ids from the HTML content.
E.g. https://music.apple.com/de/playlist/pop-rewind/pl.f642b91321c94b8d995689d0651cc2c6 with CSS selector
```.songs-list > div > div:nth-child(2) > div:nth-child(3) > div:nth-child```

We could also use a feed from https://rss.applemarketingtools.com/ to generate content for a room.

You can see a preview of my new room at https://binb.jpautz.de/gardenmix

Noutbuk committed:
Run minifiy at build time

Noutbuk committed:
Fix artists

Noutbuk committed:
Add new complex load function

Noutbuk committed:
Update node version in Dockerfile

Noutbuk committed:
Add room clearing

Noutbuk committed:
Add logging and perform refactoring

Noutbuk committed:
Fix complex config loading

Noutbuk committed:
remove rooms without song configuration

Noutbuk committed:
Fix songId generation

Noutbuk committed:
Fix song insert

Noutbuk committed:
Avoid legacy option

Noutbuk committed:
Move config to main config.json and add new kates-pop room

Noutbuk committed:
Mirgrate old room configuration

Noutbuk committed:
Fix song ids

Noutbuk committed:
Rename import data script

kschmitz123 committed:
add new room "Classics"

Noutbuk committed:
Merge pull request #1 from Noutbuk/add-new-room

add new room "Classics"

kschmitz123 committed:
remove playlists (#2)

* remove playlists

* remove rooms from list

Noutbuk committed:
Remove classics room

Test if this room causes the startup issues

Noutbuk committed:
Readd classics room

Remove query parameter from playlist urls

Noutbuk committed:
Fix leaderboard, signup and password change&reset issues

These operations used deprecated redis functions.
Additionally, the correct leadboard code was disabled.

Noutbuk committed:
Update canvas

Attempt to fix captch rendering issues

Noutbuk committed:
Fix captcha rendering

Font previously used is was not available in used Docker base image.

Noutbuk committed:
Attemp to fix captcha

Noutbuk committed:
Fix dependency

Noutbuk committed:
Fix dependencies

Noutbuk committed:
Format with prettier

Noutbuk committed:
Fix password change and password reset

Noutbuk committed:
Log password reset url to console

Sending emails too complicated at the moment.

Noutbuk committed:
Update password recovery page text
